### PR TITLE
Added fota bootloader for MIMXRT1050_EVK target

### PR DESCRIPTION
**Pull request by TaniaMirzin**

### CODE REVIEW

File: presets/TARGET_MIMXRT1050_EVK/TARGET_BL_INTERNAL_FLASH/mbed_lib.json.template [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{{
+    "name": "mbed-bootloader-{target}-{preset}",
+    "target_overrides": {{
+        "*": {{
+            "target.macros_add": [
+                "MBED_CONF_STORAGE_STORAGE_TYPE=TDB_INTERNAL",
+                "MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_BASE_ADDRESS=(MBED_ROM_START + 64*1024 + 1984*2*1024 + 64*1024)",
```
Comment (teetak01) [inline presets/TARGET_MIMXRT1050_EVK/TARGET_BL_INTERNAL_FLASH/mbed_lib.json.template:7]:
```suggestion
                "MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_BASE_ADDRESS=(MBED_ROM_START + 64*1024 + 1984*2*1024)",
```
? Is there double counting of the Bootloader ?

Comment (TaniaMirzin) [reply presets/TARGET_MIMXRT1050_EVK/TARGET_BL_INTERNAL_FLASH/mbed_lib.json.template:7]: @teetak01  indeed there is a gap between candidate storage and kvstore. But as we talked these optimization are not needed, we have here 8Mb memory of external flash that treaded as internal flash

**Review by jenia81 — APPROVED** [review]

File: presets/TARGET_MIMXRT1050_EVK/TARGET_BL_INTERNAL_FLASH/mbed_app.json [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                "The configration supports MIMXRT1050_EVK 8MB QSPI flash in default.",
+                "With the FLASHIAP and the default flash memory map, it can support",
+                "kvstore and OTA features.",
+
+                "For the Hyper flash, please delete line target.components_remove: [HYPERFLASH_BOOT]",
+                "and adjust the other settings to match the Hyper flash size and flash map.",
+                "1. Bootloader - 64kb from the beginning of flash",
+                "2. Active App Metadata Header - 512B from the end of Bootloader",
+                "3. Active App - From end of header to the 2048kb offset",
+				"4. Firmware Candidate Storage - 1984kb from the 2048kb offset",
```
Comment (jenia81) [inline presets/TARGET_MIMXRT1050_EVK/TARGET_BL_INTERNAL_FLASH/mbed_app.json]: please fix the indentation here

Comment (TaniaMirzin) [reply presets/TARGET_MIMXRT1050_EVK/TARGET_BL_INTERNAL_FLASH/mbed_app.json]: done

**Pull request merged**